server-utils/src/reactor.rs

```diff
@@ -163,4 +172,10 @@ impl RpcEventLoop {
 			warn!("Event Loop is already finished. {:?}", e);
 		});
 	}
+
+	/// Returns the close signal that can be used to close the event loop even while
+	/// another thread is blocked on the event loop in "wait"
+	pub fn close_handle(&mut self) -> Option<futures::Complete<()>>{
```

Comment:
This function doesn't make much sense - the whole point of handle is to be able to have multiple instance of it. 
This function here will `take()` the close handle and only return it for the first caller. Subsequent calls will always yield `None`.
What's more it violates the assumption in the rest of this struct (see `expect("..")` calls) - so every other function like `close()` or `wait()` will actually panic.

I still believe that having this function is completely unnecessary. See how it's done in `WebSockets` server - we own `Executor` via `Arc<Mutex<Option<Executor>>>` and close it completely inside `CloseHandle` (actually it's not perfect there, cause cloning the handle and calling `close()` on a clone would panic, but it's a separate issue).

Reply:
The WebSockets server has a separate handle that it joins on in wait(), whereas the wait() implementation in this module requires access to the executor, so we wouldn't be able to call close() on the close handle while another thread is blocked on wait(). Is there a way to construct a similar, separate handle as exists in the WebSockets implementation?